From 4247dfee7f6bcbb8b4648edba091d71adbb11ce4 Mon Sep 17 00:00:00 2001
From: Akhilesh Samineni <akhilesh.samineni@broadcom.com>
Date: Mon, 5 Apr 2021 13:27:38 -0700
Subject: [PATCH] VRF interface lookup was still done in the default-vrf which
 was causing the interface lookup to fail. Due to this static-route pointing
 to link-local was not getting installed.

---
 staticd/static_zebra.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/staticd/static_zebra.c b/staticd/static_zebra.c
index 3f3117752..2f1ef99c2 100644
--- a/staticd/static_zebra.c
+++ b/staticd/static_zebra.c
@@ -47,7 +47,7 @@
 struct zclient *zclient;
 static struct hash *static_nht_hash;
 
-static struct interface *zebra_interface_if_lookup(struct stream *s)
+static struct interface *zebra_interface_if_lookup(struct stream *s, vrf_id_t vrf_id)
 {
 	char ifname_tmp[INTERFACE_NAMSIZ];
 
@@ -55,7 +55,7 @@ static struct interface *zebra_interface_if_lookup(struct stream *s)
 	stream_get(ifname_tmp, s, INTERFACE_NAMSIZ);
 
 	/* And look it up. */
-	return if_lookup_by_name(ifname_tmp, VRF_DEFAULT);
+	return if_lookup_by_name(ifname_tmp, vrf_id);
 }
 
 /* Inteface addition message from zebra. */
@@ -120,7 +120,7 @@ static int interface_state_up(int command, struct zclient *zclient,
 {
 	struct interface *ifp;
 
-	ifp = zebra_interface_if_lookup(zclient->ibuf);
+	ifp = zebra_interface_if_lookup(zclient->ibuf, vrf_id);
 
 	if (ifp) {
 		if (if_is_vrf(ifp)) {
-- 
2.18.0

